src/components/Board/EditToolbar/EditToolbar.component.js

```diff
@@ -135,7 +137,7 @@ function EditToolbar({
           })}
           onClick={onBoardTitleClick}
         >
-          {board.name}
+          {board.name || intl.formatMessage({ id: board.nameKey })}
```

Comment:
Maybe extract this logic to a helper like `resolveLabel`, perhaps `resolveBoardName`

Reply:
Like that!